src/tensors/gpu/algorithm.h

```diff
@@ -5,12 +5,12 @@
 namespace marian {
 namespace gpu {
 template <typename T>
-void copy(Ptr<Backend> backend, const T* begin, const T* end, T* dest);
+void copy(Ptr<marian::Backend> backend, const T* begin, const T* end, T* dest);
```

Comment:
This commit allowed me to rollback some previous changes. The `cpu::Backend` and `gpu::Backend` classes were renamed to `cpu::CpuBackend` and `gpu::GpuBackend` because of a compile error related to this function. The compiler was confused between the `Backend` and the `gpu::Backend` classes when instantiating this function. This change alone fix the name ambiguity for the compiler and I renamed the classes back to their original names.